native/src/menu.rs

```diff
@@ -0,0 +1,81 @@
+//! Build menus for your application.
+use crate::keyboard::Hotkey;
+
+/// Menu representation.
+///
+/// This can be used by `shell` implementations to create a menu.
+#[derive(Debug, Clone, PartialEq)]
+pub struct Menu<Message> {
+    items: Vec<MenuEntry<Message>>,
+}
+
+impl<Message> Menu<Message> {
+    /// Creates an empty [`Menu`].
+    pub fn new() -> Self {
+        Menu { items: Vec::new() }
+    }
+
+    /// Adds an item to the [`Menu`].
+    pub fn item<S: Into<String>>(
+        mut self,
+        content: S,
+        hotkey: impl Into<Option<Hotkey>>,
+        on_activation: Message,
```

Comment:
Currently there's no implementation of the menu activation because we don't want to add `Message: Clone` to `Application`, maybe we should use `fn() -> Message` (even though the user would have to do `Menu::item(.., || Message::Example)`)?

Reply:
I just went ahead and added the `Clone` bound to `Message` in 5df2a92f28be1d53d32d5b42a6645459b7d78efe.

I know some users will be bitten by this, but I really think all messages should be cloneable and they should always represent and contain pure data.